### PR TITLE
[FIX] Fixed processing empty values of the datatable

### LINKED ISSUE #241: Loading a table with empty values at the end leads to `undefined` missing values
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Expected behavior

If I load a table with - in an extreme case - all missing values, I would expect to 
- see only one missing value in the annotation tool and
- that missing value to be written out as `""`, i.e. empty string in the data dictionary

### Current behavior

If I have only missing values in a row, then they will show up as `undefined` instead of `""`.

Specifically, this is what we see:

- if the empty value is followed by more empty values in the same row -> `undefined`
- if the empty value is followed by at least one more nonempty value (going left to right), -> `""`

### Error message

the first time this gets flagged is by the schema validation of the output data dictionary, which does not allow a missing value to be `undefined`

### Environment

_No response_

### How to reproduce

1. Take a table with all empty rows but more than 2 columns
2. Annotate column 1 as "about" e.g. "sex"
3. Mark the "empty" value in the value annotation as missing
4. Notice the missing value in the output data dictionary preview show up as `undefined` and the schema validation failing

Variant: 
- if you have a table where 
  - column 2 has all empty values and
  - column 3 has some empty and some non-empty values
- THEN: when you annotate column 2, you will see two missing values and
- in the output you will see both `undefined` and `""` as missing values

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #241 

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass
- [ ] If the PR changes the default [Neurobagel `config.json` or any of the term files](https://github.com/neurobagel/communities/tree/main/configs/Neurobagel), make sure their respective counterparts in [neurobagel/communities](https://github.com/neurobagel/communities) have been updated accordingly.

For new features:
- [ ] Tests have been added

For bug fixes:
- [x] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Extract TSV parsing into a dedicated utility, add tests for empty value handling, and update the data store to use this function to correctly process empty TSV fields

New Features:
- Introduce parseTsvContent utility to extract TSV parsing logic

Bug Fixes:
- Preserve empty fields when parsing TSV content for the data table

Enhancements:
- Replace inline TSV parsing in the data store with parseTsvContent

Tests:
- Add unit tests for parseTsvContent to verify handling of empty values

### CODE REVIEW

File: src/utils/util.test.ts [hunk 1]
```diff
@@ -2,12 +2,33 @@ import axios from 'axios';
...
+  fetchConfig,
+  mapConfigFileToStoreConfig,
+} from './util';
 
 // Mock axios
 vi.mock('axios');
 const mockedAxios = vi.mocked(axios, true);
 
+describe('parseTsvContent', () => {
+  it('should parse TSV content correctly', () => {
```
Comment (surchs) [inline src/utils/util.test.ts:17]:
```suggestion
  it('parses rows of empty values correctly as string', () => {
```
